src/zenml/integrations/kubernetes/metadata_stores/kubernetes_metadata_store.py

```diff
@@ -0,0 +1,102 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of Kubernetes metadata store."""
+
+from typing import ClassVar
+
+from kubernetes import client, config
+
+from zenml.integrations.kubernetes import KUBERNETES_METADATA_STORE_FLAVOR
+from zenml.integrations.kubernetes.orchestrators.kube_utils import (
+    create_mysql_deployment,
+    create_namespace,
+    delete_deployment,
+    make_core_v1_api,
+)
+from zenml.logger import get_logger
+from zenml.metadata_stores.mysql_metadata_store import MySQLMetadataStore
+
+logger = get_logger(__name__)
+
+
+class KubernetesMetadataStore(MySQLMetadataStore):
```

Comment:
If I understood this metadata store correctly, it's just a wrapper around the MySQL metadata store in order to deploy it to a k8s cluster right?

I was thinking about ways to make that a little clearer, as people who just want to connect to a MySQL database running in kubernetes shouldn't register a metadata store of flavor `kubernetes` I think. They would just run into trouble if other components of their stack require a `zenml stack up` (e.g. the KubeflowOrchestrator) in which case they'll get errors about being unable to provision the metadata store in their cluster (Or even worse, we unknowingly provision the cluster for them). 

The only thing I could think of was making the attributes like `kubernets_namespace` and `deployment_name` either optional (in which case we just wouldn't provision anything) or without default (this at least forces them to think about what they're registering). Any thoughts?

Reply:
Yeah you understood it right, it's just a wrapper around MySQL that deploys the database into k8s. I like both of your suggestions, personally, I'm leaning towards removing the defaults. Might also make sense to print a custom error when people try to register it without those args, which explains the use case of this metadata store and recommends people to use MySQL instead if they already have a DB running somewhere. Would that make it clear, what do you think?

Reply:
Yep I like that idea!

Reply:
Cool, changed it accordingly.